tests/lib/rules/prop-types.js

```diff
@@ -1361,6 +1361,30 @@ ruleTester.run('prop-types', rule, {
       ].join('\n'),
       options: [{skipUndeclared: false}],
       parserOptions: parserOptions
+    }, {
+      // Async functions can't be components.
+      code: [
+        'var Hello = async function(props) {',
+        '  return <div>Hello {props.name}</div>;',
+        '}'
+      ].join('\n'),
+      parser: 'babel-eslint'
```

Comment:
there's no need to use babel-eslint here - let's test this with eslint's default parser, which fully support `async function` now.